shell/common/shell.h

```diff
@@ -34,8 +34,43 @@
 
 namespace flutter {
 
-/// Wraps up all the different components of Flutter engine and coordinates them
-/// through a series of delegates.
+//------------------------------------------------------------------------------
+/// Perhaps the single most important class in the Flutter engine repository.
+/// When embedders create a Flutter application, they are referring to the
+/// creation of an instance of a shell. Creation and destruction of the shell is
+/// synchronous and the embedder only holds a unique pointer to the shell. The
+/// shell does not create the threads its primary components run on. Instead, it
+/// is embedder responsibility to create threads and give the shell task runners
+/// for those threads. Due to deterministic destruction of the shell, the
+/// embedder can terminate all threads immediately after collecting the shell.
+/// The shell must be created and destroyed on the same thread but different
+/// shells (i.e. a separate instance of a Flutter application) may be run on
+/// different threads simultaneously. The task runners themselves do not have to
+/// be unique. If all task runner references given to the shell during shell
+/// creation point to the same task runner, the Flutter application is
+/// effectively single threaded.
+///
+/// The shell is the central nervous system of the Flutter application. None of
+/// the shell components are thread safe and may only be created, accessed and
```

Comment:
maybe: "None of the shell components are thread safe and must be created, accessed and destroyed on the same thread"

Reply:
Done.